### ISSUE TITLE
Comparison of modulo result is always true in function draw_cur_mob

### ISSUE THREAD

**Issue by petervas** [NagyD/SDLPoP#66]
The following code from seg007.c produces a warning in cppcheck:
```
// seg007:132C
void __pascal far draw_cur_mob() {
	if (curmob.room == drawn_room) {
		redraw_height = 0x20;
		set_redraw_full(curr_tilepos, 1);
		set_wipe(curr_tilepos, 1);
		if ((curr_tilepos + 1) % 10 < 10) {
			set_redraw_full(curr_tilepos + 1, 1);
			set_wipe(curr_tilepos + 1, 1);
		}
	}
}
```

Specifically the check
```
if ((curr_tilepos + 1) % 10 < 10)
```
will always be true. 

Did you mean to check for something else?

**Comment by NagyD** [comment]
Nice find!

It seems to me that the original game does the same, however:
(You can find a link to the full disassembly in the Readme.)
```
seg007:1358  A0 ED 42  mov  al, curr_tile_pos
seg007:135B  2A E4     sub  ah, ah
seg007:135D  8B F0     mov  si, ax
seg007:135F  46        inc  si
seg007:1360  8B C6     mov  ax, si
seg007:1362  2B D2     sub  dx, dx
seg007:1364  B9 0A 00  mov  cx, 10
seg007:1367  F7 F1     div  cx
seg007:1369  3B D1     cmp  dx, cx
seg007:136B  73 18     jnb  loc_98D5
```
(Why does it bother `si`, by the way?)

My guess is that they meant `if ((curr_tilepos % 10) + 1 < 10)` or `if (curr_tilepos % 10 < 9)`.

The intended purpose of the check is that a loose floor landing in the rightmost column should not redraw the tile on the right.

To test this, switch to blind mode (Shift+B), and make a loose floor land in the rightmost column.
Level 11 is perfect for this.
With the bug, the first tile of the next row is redrawn for a single frame.

**Comment by NagyD** [comment]
Fixed: 00c5f5184886baf9197f5022af06f6fbd4fc06ae
